### ISSUE TITLE
Support kubernetes

### ISSUE THREAD

**Issue by amouat** [amouat/reggie#1]
Current PoC only supports swarm, but it should be easy to get working on k8s. First step is figuring out how to get list of peers.

**Comment by amouat** [comment]
Probably the best way to do this is to use a daemon set and an image wolf namespace. Then we can just do a GET request for the pods in the namespace.

**Comment by amouat** [comment]
Actually, the downward API may be the easiest way to get the pod info, if it has the needed data https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#the-downward-api

I suspect it only exposes info about the given pod though :(

**Comment by etiennetremel** [comment]
Using DaemonSet and headless Service allow proper peer discovery: https://github.com/ContainerSolutions/ImageWolf/pull/6

**Comment by amouat** [comment]
Thanks Etienne! Just taking a look at this now.

**Comment by etiennetremel** [comment]
If you hit some version issue between Docker client and the api, you need to downgrade the Docker client version. I had that issue while testing on GKE. Using the following multistage Dockerfile fixed it:

```
# Build stage
FROM golang:1.8 as builder

WORKDIR /go/src/

COPY src .

RUN go get github.com/anacrolix/torrent
RUN go get github.com/anacrolix/utp
RUN go get github.com/docker/distribution/notifications

RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o app .

# Run stage
FROM alpine:3.4
RUN apk --no-cache add ca-certificates docker=1.11.2-r1

WORKDIR /root/

COPY --from=builder /go/src/app .

EXPOSE 6000

CMD ["./app"]
```

**Comment by amouat** [comment]
Closed via #7
